ChaosMod/Effects/db/Vehs/VehsRandomTraffic.cpp

```diff
@@ -0,0 +1,52 @@
+#include <stdafx.h>
+//effect by ProfessorBiddle, this is a bit clunky but it works well.
+static void OnTick()
+{
+	Vehicle playerVeh = GET_VEHICLE_PED_IS_IN(PLAYER_PED_ID(), false);
+	for (auto veh : GetAllVehs())
+	{
+		//softlock prevention, hopefully
+		if (IS_ENTITY_A_MISSION_ENTITY(veh)) { continue; }
+
+		//detect if vehicle already randomized
+		auto carModified = GET_VEHICLE_WINDOW_TINT(veh);
+
+		if(veh != playerVeh && (carModified != 3))
+		{
+			//remove ped from vehicle and place them out of the way for now
+			Ped pedInCar = GET_PED_IN_VEHICLE_SEAT(veh, -1, 0);
+			Vector3 pedCoords = GET_ENTITY_COORDS(pedInCar, false);
+			SET_ENTITY_COORDS(pedInCar, pedCoords.x, pedCoords.y, pedCoords.z + 30.f, 1, 0, 0, 1);
+
+			//spawn random vehicle in place of old vehicle
+			Vector3 prevPos = GET_ENTITY_COORDS(veh, false);
+			float prevRot = GET_ENTITY_HEADING(veh);
+			SET_ENTITY_AS_MISSION_ENTITY(veh, true, true);
+			DELETE_VEHICLE(&veh);
+			static std::vector<Hash> vehModels = Memory::GetAllVehModels();
+			if (!vehModels.empty())
+			{
+				Vehicle veh = CreatePoolVehicle(vehModels[g_random.GetRandomInt(0, vehModels.size() - 1)], prevPos.x, prevPos.y, prevPos.z, prevRot);
```

Comment:
Use `CreateTempVehicle` if possible to avoid despawning other entities added to the pool by the mod (since it'll easily hit the 30 entities limit).